### PR TITLE
Fix /auth compat endpoint

### PR DESCRIPTION
**Pull request by matejvasek**
Fix `/auth` Docker APIv2 endpoint to match `moby` behavior.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM
@jwhonce PTAL

**Comment by matejvasek** [comment]
I need to update/create tests.

File: pkg/api/handlers/compat/auth.go [hunk 1]
```diff
@@ -50,9 +50,19 @@ func Auth(w http.ResponseWriter, r *http.Request) {
 			Status:        "Login Succeeded",
 		})
 	} else {
-		utils.WriteResponse(w, http.StatusBadRequest, entities.AuthReport{
-			IdentityToken: "",
-			Status:        "login attempt to " + authConfig.ServerAddress + " failed with status: " + err.Error(),
+		var msg string
+
+		var unauthErr DockerClient.ErrUnauthorizedForCredentials
+		if errors.As(err, &unauthErr) {
```
Comment (Luap99) [inline pkg/api/handlers/compat/auth.go:56]: I think this should use `errors.Is`

Comment (matejvasek) [reply pkg/api/handlers/compat/auth.go:56]: @Luap99
AFAIK `Is()` is used when checking against instance (sentinel error) and `As()` when checking against type. I ma checking against type.

Comment (matejvasek) [reply pkg/api/handlers/compat/auth.go:56]: ErrUnauthorizedForCredentials is `type` not `var`

Comment (Luap99) [reply pkg/api/handlers/compat/auth.go:56]: ok, sorry for the noise.

File: pkg/api/handlers/compat/auth.go [hunk 2]
```diff
@@ -50,9 +50,19 @@ func Auth(w http.ResponseWriter, r *http.Request) {
...
+			msg = "401 Unauthorized"
+		} else {
+			msg = err.Error()
+		}
+
+		utils.WriteResponse(w, http.StatusInternalServerError, struct {
```
Comment (Luap99) [inline pkg/api/handlers/compat/auth.go:62]: This should be a 401 and not 500 code when you match ErrUnauthorizedForCredentials

Comment (Luap99) [reply pkg/api/handlers/compat/auth.go:62]: Otherwise you return 500 but it says 401 in the error message

Comment (matejvasek) [reply pkg/api/handlers/compat/auth.go:62]: @Luap99 totally agree but `moby` returns `500` so just to be sure I return the same.

Comment (Luap99) [reply pkg/api/handlers/compat/auth.go:62]: Alright, bug for bug compatibility then

**Review by Luap99 — COMMENTED** [review]
LGTM

File: pkg/api/handlers/compat/auth.go (hunk 2 above)
Comment (matejvasek) [reply pkg/api/handlers/compat/auth.go:62]: also returning `401` would make this code work:
```go
	_, err = cli.RegistryLogin(ctx, types.AuthConfig{Username: username, Password: password, ServerAddress: registry})
	if dockerClient.IsErrUnauthorized(err) {
		return ErrUnauthorized
	}
```
right now I have to do this awful awful thing:
```go
	_, err = cli.RegistryLogin(ctx, types.AuthConfig{Username: username, Password: password, ServerAddress: registry})
	if err != nil && strings.Contains(err.Error(), "401 Unauthorized") {
		return ErrUnauthorized
	}
```
Still I am afraid doing something differently than `moby`.

Comment (matejvasek) [reply pkg/api/handlers/compat/auth.go:62]: I'll add comment to src code.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
/lgtm
/hold
#until tests pass

**Pull request merged**